builder/job.go

```diff
@@ -121,27 +127,47 @@ func Build(d *daemon.Daemon, buildConfig *Config) error {
 	} else if urlutil.IsURL(buildConfig.RemoteURL) {
 		f, err := httputils.Download(buildConfig.RemoteURL)
 		if err != nil {
-			return err
+			return fmt.Errorf("Error downloading remote context %s: %v", buildConfig.RemoteURL, err)
 		}
 		defer f.Body.Close()
-		dockerFile, err := ioutil.ReadAll(f.Body)
+		ct := f.Header.Get("Content-Type")
+		clen := int(f.ContentLength)
+		contentType, bodyReader, err := inspectResponse(ct, f.Body, clen)
```

Comment:
not 100% sure but do you need to add
```
defer bodyReader.close()
```
after this line.  I know there's a f.Body.Close() above, but will that automatically close any readers you created in inspectResponse()?  I'd hope so but not sure.